health/server.go

```diff
@@ -110,7 +117,14 @@ func (s *Server) Watch(in *healthpb.HealthCheckRequest, stream healthgrpc.Health
 func (s *Server) SetServingStatus(service string, servingStatus healthpb.HealthCheckResponse_ServingStatus) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	if s.shutdownEvent.HasFired() {
```

Comment:
I think this could go above the lock.  (?)

Reply:
There can be a race if this is not in the lock

goroutine-1: `HasFired()=false`
goroutine-2: Fire()
goroutine-2: take lock, set Not_serving, release lock
goroutine-1: take lock, set status (to Serving)